### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by Progress2002**
**Members: Progress2002 & Kodjo Matthieu SENOU**

**The following was archived in this milestone:**

- [x]  Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x]  Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

clinic_diagram

- [x] Create a file named schema_based_on_diagram.sql where you implement the database from the diagram above.
- [x] Join tables from many-to-many relationships might not appear in the diagram, but you still need them.
- [x] Remember to add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "invoice_id" INT,
+  "treatment_id" INT,
+  PRIMARY KEY ("id"),
+  CONSTRAINT "FK_invoice_items.treatment_id"
+    FOREIGN KEY ("treatment_id")
+      REFERENCES "treatments"("id"),
+  CONSTRAINT "FK_invoice_items.invoice_id"
+    FOREIGN KEY ("invoice_id")
+      REFERENCES "invoices"("id")
+);
```
Comment (afanimartin) [inline schema_based_on_diagram.sql:57]: - Please a table to handle `many-to-many` relationship between `Treatments` and `Medical_histories` tables. You can do something like:
```ruby
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
+-- create medical database
+CREATE DATABASE medical
```
Comment (afanimartin) [inline schema_based_on_diagram.sql:2]: - Please add indexes for columns that are mostly used. Something like below will do;
```sql
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Progress2002, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Required tables created with columns
- Readme template is professional 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @Progress2002 and @Matthieu96Code 👏 👏,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Create a file to implement the given database from the diagram ✔️ 
Create Join tables from many-to-many relationships ✔️ 
Add the FK indexes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Progress2002) [reply schema_based_on_diagram.sql:57]: @afanimartin Thanks for your observation and suggestion. The file has been updated.

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Progress2002) [reply schema_based_on_diagram.sql:2]: @afanimartin Thanks for the well-detailed review. Changes have been made to meet the requirements.

**Pull request merged**